Common/src/Pollster/Agent.cs

```diff
@@ -47,16 +48,18 @@ namespace ArmoniK.Core.Common.Pollster;
 /// </summary>
 public sealed class Agent : IAgent
 {
-  private readonly List<TaskCreationRequest>                  createdTasks_;
-  private readonly ILogger                                    logger_;
-  private readonly IObjectStorage                             objectStorage_;
-  private readonly IPushQueueStorage                          pushQueueStorage_;
-  private readonly IResultTable                               resultTable_;
-  private readonly Dictionary<string, (byte[] id, long size)> sentResults_;
-  private readonly SessionData                                sessionData_;
-  private readonly ISubmitter                                 submitter_;
-  private readonly TaskData                                   taskData_;
-  private readonly ITaskTable                                 taskTable_;
+  private readonly ConcurrentBag<Result>                              createdResults_;
+  private readonly ConcurrentBag<TaskCreationRequest>                 createdTasks_;
```

Comment:
I think those could be `ConcurrentBag<List<?>>` and flattened in the post processing to reduce the cost of adding multiple elements inside the bag and iterating over it.